iam-login-service/src/main/java/it/infn/mw/iam/core/oauth/IamDeviceEndpointController.java

```diff
@@ -257,6 +261,10 @@ public String approveDevice(@RequestParam("user_code") String userCode,
 
     model.put("approved", true);
 
+    Date timeout = null;
+    approvedSiteService.createApprovedSite(client.getClientId(), auth.getName(), timeout,
```

Comment:
The problem here is the case where there is a scope policy. Since it is evaluated after this step, in the approved sites we will see also the filtered scope - and in fact we see it on the consent page and actually approve it, even though it is then removed from the scope policy filter in the access token. 
In generale, this mitre code needs to be improved later!